### PR TITLE
3.x: Refactor to io.reactivex.rxjava3.**

### PR DESCRIPTION
**Pull request by akarnokd**
This PR refactors the package structure to match the group ID: `io.reactivex.rxjava3`. In addition, the base classes and interfaces now live in `io.reactivex.rxjava3.core`.

In addition, the Reactive Streams dependency was bumped to 1.0.3-RC1 to verify the TCK changes.

### CODE REVIEW

**Review by vanniktech — APPROVED** [review]
I trust you

**Comment by akarnokd** [comment]
You can compare lines and branch counts: https://github.com/ReactiveX/RxJava/pull/6620#issuecomment-521889856

```diff
@@             Coverage Diff              @@
##                3.x    #6620      +/-   ##
============================================
+ Coverage     98.13%   98.17%   +0.03%     
- Complexity     6187     6188       +1     
============================================
  Files           678      678              
  Lines         44792    44792              
  Branches       6173     6173              
============================================
+ Hits          43956    43973      +17     
+ Misses          290      281       -9     
+ Partials        546      538       -8
```

**Comment by vanniktech** [comment]
Go ahead and merge!

File: build.gradle [hunk 1]
```diff
@@ -52,7 +52,7 @@ targetCompatibility = JavaVersion.VERSION_1_6
 // ---------------------------------------
 
 def junitVersion = "4.12"
-def reactiveStreamsVersion = "1.0.2"
+def reactiveStreamsVersion = "1.0.3-RC1"
```
Comment (artem-zinnatullin) [inline build.gradle:55]: Prob should have been done in a separate PR, but not a big deal

**Review by artem-zinnatullin — COMMENTED** [review]
👍
